test/mocks/server/BUILD

```diff
@@ -198,8 +198,14 @@ envoy_cc_mock(
 
 envoy_cc_mock(
     name = "server_factory_context_mocks",
-    srcs = ["server_factory_context.cc"],
-    hdrs = ["server_factory_context.h"],
+    srcs = [
+        "server_factory_context.cc",
+        "transport_socket_factory_context.cc",
```

Comment:
this is to avoid circular dependencies between both mocks. probably it should be only one.